src/wrapper.jl

```diff
@@ -106,22 +214,952 @@ const FT_PURGE_TX = 2
+end
 
-@enum(
-  FTParity,
-  PARITY_NONE = FT_PARITY_NONE,
-  PARITY_ODD  = FT_PARITY_ODD,
-  PARITY_EVEN = FT_PARITY_EVEN,
-  PARITY_MARK = FT_PARITY_MARK,
-  PARITY_SPACE = FT_PARITY_SPACE)
-
-  
\ No newline at end of file
+"""
+    mutable struct FT_HANDLE<:IO
+
+Holds a handle to an FT D2XX device.
+"""
+mutable struct FT_HANDLE<:IO
+  p::Ptr{Cvoid} 
+end
+
+"""
+    FT_HANDLE()
+
+Constructs a handle to an FT D2XX device that points to `C_NULL`. Initialsed 
+with a finalizer that calls [`destroy!`](@ref).
+"""
+function FT_HANDLE()
+  handle = FT_HANDLE(C_NULL)
+  @compat finalizer(destroy!, handle)
+  handle
+end
+
+"""
+    destroy!(handle::FT_HANDLE)
+
+Destructor for the [`FT_HANDLE`](@ref) type.
+"""
+function destroy!(handle::FT_HANDLE)
+  if ptr(handle) != C_NULL
+    FT_Close(handle)
+  end
+end
+
+"""
+    ptr(handle::FT_HANDLE)
+
+Get the raw pointer for an [`FT_HANDLE`](@ref).
+"""
+ptr(handle::FT_HANDLE) = handle.p
+
+"""
+    ptr(handle::FT_HANDLE, fthandle_ptr::Ptr{Cvoid})
+
+Set the raw pointer for an [`FT_HANDLE`](@ref).
+"""
+ptr(handle::FT_HANDLE, fthandle_ptr::Ptr{Cvoid}) = (handle.p = fthandle_ptr)
+
+# wrapper functions
+#
+
+"""
+    FT_CreateDeviceInfoList()
+
+Wrapper for D2XX library function `FT_CreateDeviceInfoList`. 
+
+See D2XX Programmer's Guide (FT_000071) for more information.
+
+# Example
+```julia-repl
+julia> numdevs = FT_CreateDeviceInfoList()
+0x00000004
+
+julia> "Number of devices is \$numdevs"
+"Number of devices is 4"
+```
+"""
+function FT_CreateDeviceInfoList()
+  lpdwNumDevs = Ref{DWORD}(0)
+  status = ccall(cfunc[:FT_CreateDeviceInfoList], cdecl, FT_STATUS, 
+                 (Ref{DWORD},),
+                 lpdwNumDevs)
+  FT_STATUS_ENUM(status) == FT_OK || throw(FT_STATUS_ENUM(status))
+  lpdwNumDevs[]
+end
+
+
+
+"""
+    FT_GetDeviceInfoList(lpdwNumDevs)
+
+Wrapper for D2XX library function `FT_GetDeviceInfoList`. 
+
+See D2XX Programmer's Guide (FT_000071) for more information.
+
+# Arguments
+ - `lpdwNumDevs`: The number of devices.
+
+# Example
+```julia-repl
+julia> numdevs = FT_CreateDeviceInfoList()
+0x00000004
+
+julia> devinfolist, numdevs = FT_GetDeviceInfoList(numdevs);
+
+julia> numdevs
+0x00000004
+
+julia> using LibFTD2XX.Util # for ntuple2string
+
+julia> ntuple2string(devinfolist[1].description)
+"USB <-> Serial Converter D"
+
+julia> devinfolist[1].fthandle_ptr
+Ptr{Nothing} @0x0000000000000000
+
+julia> devinfolist[1].locid
+0x00000000
+
+julia> devinfolist[1].typ
+0x00000007
+
+julia> devinfolist[1].flags
+0x00000002
+
+julia> devinfolist[1].id
+0x04036011
+
+julia> ntuple2string(devinfolist[1].serialnumber)
+"FT3AD2HCD"
+
+```
+"""
+function FT_GetDeviceInfoList(lpdwNumDevs)
+  pDest =  @compat Vector{FT_DEVICE_LIST_INFO_NODE}(undef, lpdwNumDevs)
+  status = ccall(cfunc[:FT_GetDeviceInfoList], cdecl, FT_STATUS, 
+                 (Ref{FT_DEVICE_LIST_INFO_NODE}, Ref{DWORD}),
+                  pDest,                         Ref{DWORD}(lpdwNumDevs))
```

Comment:
No throw?

Reply:
Good spot! Fixed in dac0d67c7c433682b99b2eb0f1d0d59c3c45334c